plugins/plugin_utils/connectivity_utils.py

```diff
@@ -0,0 +1,373 @@
+            _create_service_configuration(rest_client, resource_uri, port)
+            new_service_config = True
+            try:
+                cred = _list_credentials(rest_client, resource_uri, certificate_validity_in_seconds)
+            except Exception as e:
+                raise AnsibleError(f"Unable to get relay information. Failed with error: {str(e)}")
+            _handle_relay_connection_delay(rest_client, "Setting up service configuration")
+    return (cred, new_service_config)
+
+
+def _check_service_configuration(rest_client, resource_uri, port):
+    url = f"/{resource_uri}/providers/Microsoft.HybridConnectivity/endpoints/default/serviceConfigurations/SSH"
+
+    serviceConfig = None
+    # pylint: disable=broad-except
+    try:
+        serviceConfig = resource(rest_client, url, "GET")
+    except Exception:
+        # If for some reason the request for Service Configuration fails,
+        # we will still attempt to get relay information and connect. If the service configuration
+        # is not setup correctly, the connection will fail.
+        # The more likely scenario is that the request failed with a "Authorization Error",
+        # in case the user isn't an owner/contributor.
+        return True
+    serviceConfigPort = serviceConfig and serviceConfig.get('properties', {}).get('port')
+    if port:
+        return int(serviceConfigPort) == int(port)
+
+    return True
+
+
+def _create_default_endpoint(rest_client, resource_uri):
+    url = f"/{resource_uri}/providers/Microsoft.HybridConnectivity/endpoints/default"
+    body = {'properties': {'type': 'default'}}
+
+    hostname = parse_resource_id(resource_uri)["name"]
+    resource_group = parse_resource_id(resource_uri)["resource_group"]
+    try:
+        endpoint = resource(rest_client, url, "PUT", body=body)
+    except HttpResponseError as e:
+        if e.reason == "Forbidden":
+            raise AnsibleError(f"Client is not authorized to create a default connectivity "
+                               f"endpoint for \'{hostname}\' in Resource Group \'{resource_group}\'. "
+                               f"This is a one-time operation that must be performed by "
+                               f"an account with Owner or Contributor role to allow "
+                               f"connections to the target resource.")
+        raise AnsibleError(f"Failed to create default endpoint for the target Arc Server. "
+                           f"\nError: {str(e)}")
+    except Exception as e:
+        raise AnsibleError(f"Failed to create default endpoint for the target Arc Server. "
+                           f"\nError: {str(e)}")
+
+    return endpoint
+
+
+def _create_service_configuration(rest_client, resource_uri, port):
+    if not port:
+        port = '22'
+
+    url = f"/{resource_uri}/providers/Microsoft.HybridConnectivity/endpoints/default/serviceConfigurations/SSH"
+    body = {'properties': {'port': int(port), 'serviceName': 'SSH' }}
```

Comment:
whitespace before '}'
```suggestion
    body = {'properties': {'port': int(port), 'serviceName': 'SSH'}}
```